tasks/nginx.yml

```diff
@@ -0,0 +1,28 @@
+---
+- name: Enable network connections between nginx and Jenkins
+  seboolean:
+    name: httpd_can_network_connect
+    state: yes
+    persistent: yes
+
+- name: Check nginx redirects HTTP to HTTPS
+  uri:
+    url: http://localhost
+    follow_redirects: none
+    status_code: 301
+  register: http_req
+- name: Fail if HTTP doesn't redirect to HTTPS
+  fail:
+  when: http_req.location != 'https://localhost/'
+
+- name: Check Jenkins responds via HTTPS
+  uri:
+    url: https://localhost
+    follow_redirects: none
+    return_content: yes
+    # TODO have it use the test cert
+    validate_certs: no
+  register: https_req
+- name: Fail if HTTPS response isn't from Jenkins
+  fail:
+  when: "'Jenkins' not in https_req.content"
```

Comment:
These are examples of building [smoke tests](https://en.wikipedia.org/wiki/Smoke_testing_(software)) into roles themselves, an approach I have been very happy with.